low-level/framework/sspl_ll_d

```diff
@@ -423,37 +424,54 @@ def _tune_modules_for_env(modlist):
     # current deployment env declared here, Extend checks based on need in future
     # like server_type, etc
 
-    storage_type = "virtual"
-    server_type = "virtual"
-    server_per_node = 2
-    nodes = 1
+    NODE_KEY = 'nodes'
+    SERVER_PER_NODE_KEY = 'servers_per_node'
+    STORAGE_TYPE_KEY = 'storage_type'
+    SERVER_TYPE_KEY = 'server_type'
+
+    STORAGE_TYPE = 'physical'
+    SERVER_TYPE = 'rbod'
+    SERVER_PER_NODE = 2
+    NODES = 1
+
     info_err = None
+
     try:
-        setup_info = subprocess.Popen("sudo /usr/bin/provisioner get_setup_info", shell=True,
-                        stdout=subprocess.PIPE).communicate()[0].decode("utf-8").rstrip()
-        setup_info = ast.literal_eval(setup_info)
-        try:
-            storage_type = setup_info['storage_type'].lower()
-            server_type = setup_info['server_type'].lower()
-            server_per_node = setup_info['servers_per_node'].lower()
-            nodes = setup_info['nodes']
-            logger.info(f"sspl-ll Bootstrap: deployment environment: storage_type : {storage_type}, server_type : {server_type}")
-            logger.info(f"sspl-ll Bootstrap: deployment environment: server_per_node : {server_per_node}, nodes : {nodes}")
-        except Exception as err:
-            info_err = err
-            logger.error(f"Error in getting setup info: {err}")
+        util = Utility()
+    except Exception as err:
+        logger.error(f'Can not instantiate Utility class, Error:{err}')
 
+    try:
+        if not util:
+            # IF for any reason util class fails to instantiate, without blocking the execution
+            # Go ahead with excution using subprocess module here directly
+            process = subprocess.Popen(['sudo', 'provisioner', 'get_setup_info'], \
+                                 shell=False, stdout=subprocess.PIPE)
+            _result, _ret_code = process.communicate()[0], process.returncode
+        else:
+            _result, _err, _ret_code = util.execute_cmd(['sudo', 'provisioner', 'get_setup_info'])
+            _err = _err.decode("utf-8").rstrip()
+        if _ret_code == 0 and _err is '':
```

Comment:
**Possible exception:** In case 'if' clause(line 445) is satisfied and not entered into else, we don't have reference for _err variable at line 454.

Reply:
Right !